lib/msf/ui/console/command_dispatcher/auxiliary.rb

```diff
@@ -48,9 +55,27 @@ def method_missing(meth, *args)
 
       return mod.send(meth.to_s, *args)
     end
+
+    action = meth.to_s.delete_prefix('cmd_')
+    if mod && mod.kind_of?(Msf::Module::HasActions) && mod.actions.map(&:name).map(&:downcase).include?(action)
+       do_action(action, *args)
+    end
+
     return
   end
 
+  #
+  #
+  # Execute the module with a set action
+  #
+  def do_action(meth, *args)
+    action = mod.actions.select { |action| action.name.downcase == meth.downcase }.first
+    raise Exception('no action') if action.nil?
```

Comment:
Using the proper exception type from [Ruby exception tree](https://airbrake.io/blog/ruby-exception-handling/ruby-exception-classes) really makes error handling in a large program much clearer. The `rescue` needed to handle a base `Exception` object would also catch many kinds of exceptions that a routine error handler should not, like out of memory errors, interrupts, and syntax errors. For example, such a `rescue Exception` would have caught and handled the `NoMethodError` that `Exception('no action')` causes, where the bug could have gone unnoticed for quite some time. Those types of `rescue` patterns and the errors they were intended to catch make debugging _much_ more difficult when an unintended exception type crops up.

 Also,  a good exception message should say `what` triggered the error, if possible. It is the first thing someone debugging would have to figure out, so including it really helps anyone who comes across it in the future (often times yourself). 

```suggestion
    raise ArgumentError.new("invalid module action given: #{meth}") if action.nil?
```

Reply:
So after thinking a bit, I realized that the [Metasploit exception tree](https://github.com/rapid7/metasploit-framework/tree/master/lib/msf/core/exceptions.rb) has a custom exception for this: https://github.com/rapid7/metasploit-framework/blob/f49bf7b09a56f48a814734a4b4a3937e0765257b/lib/msf/core/exceptions.rb#L207

So this should be
```suggestion
    raise Msf::MissingActionError.new(meth) if action.nil?
```